save-backend/src/main/kotlin/com/saveourtool/save/backend/security/OrganizationPermissionEvaluator.kt

```diff
@@ -33,6 +34,23 @@ class OrganizationPermissionEvaluator(
         return lnkUserOrganizationService.findRoleByUserIdAndOrganization(userId, organization).isHigherOrEqualThan(role)
     }
 
+    /**
+     * @param authentication [Authentication] describing an authenticated request
+     * @param organization is organization in which we want to change the status
+     * @param newStatus is new status in [organization]
+     * @return whether user described by [authentication] can have permission on change [organization] status on [newStatus]
+     */
+    fun hasPermissionToChangeStatus(authentication: Authentication?, organization: Organization, newStatus: OrganizationStatus): Boolean {
+        val oldStatus = organization.status
+
+        return when {
+            oldStatus == newStatus -> false
```

Comment:
why false here?

Reply:
in this case, the status of the organization and the status that we want to set are the same and therefore the action is not necessary

Reply:
so even user has permission to do it, user gets 401. it can mislead

Reply:
Well, this operation is useless. In principle, the button to restore the organization will not be displayed on the frontend if the organization is already ``CREATED``

Reply:
Or is it better for me to allow any action, even if a person wants to do so?

Reply:
suggest to check oldStatus and newStatus in controller and return 404 to user

Reply:
Or is it better to check for statuses earlier? And in case of a match, issue a 404?

Reply:
> Or is it better to check for statuses earlier? And in case of a match, issue a 404?

yes

> Or is it better for me to allow any action, even if a person wants to do so?

nope, I think we need to check permission in all cases even old status == new status in this method.
probably we can throw exception here if we check changing permission to same value

Reply:
Done